File: LeetCode/Medium/1679-max-number-of-k-sum-pairs/1679-max-number-of-k-sum-pairs.go
func maxOperations(nums []int, k int) int {
    sort.Ints(nums)
    ret, i, j := 0, 0, len(nums) - 1
    var temp int
    
    for i < j {
        temp = nums[i] + nums[j]
        if temp == k {
            ret++
            i++
            j--
        } else if temp < k {
            i++
        } else { j-- }
    }
    
    return ret
}

File: LeetCode/Medium/1679-max-number-of-k-sum-pairs/1679-max-number-of-k-sum-pairs.cpp
class Solution {
public:
    int maxOperations(vector<int>& nums, int k) {
        sort(nums.begin(), nums.end());
        int i = 0;
        int j = nums.size() - 1;
        int ret = 0;
        
        while (i < j){
            int temp = nums[i] + nums[j];
            if (temp == k){
                ++ret;
                ++i;
                j--;
            } else if (temp < k) ++i;
            else j--;
        }
        
        return ret;
    }
};